File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/repository/flights/IFlightRepository.cs
﻿using Proiect_MPP.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.repository.flights
{
    public interface IFlightRepository : Repository<Flight, int>
    {
        List<Flight> getByDepartureAirport(int departureAirport);
        List<Flight> getByDestinationAirport(int destinationAirport);
        List<Flight> getAfterDepartureDateTime(DateTime departureDateTime);
        List<Flight> getAvailable();
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/src/main/java/root/proiect_mpp/repositories/flights/IFlightRepository.java
package root.proiect_mpp.repositories.flights;

import root.proiect_mpp.domain.Flight;
import root.proiect_mpp.repositories.Repository;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Collection;

public interface IFlightRepository extends Repository<Flight, Integer> {
    Collection<Flight> getByDepartureAirport(int departureAirport);
    Collection<Flight> getByDestinationAirport(int destinationAirport);
    Collection<Flight> getAfterDepartureDateTime(LocalDateTime departureDateTime);
    Collection<Flight> getAvailable();
}
